queries/ecma/highlights.scm

```diff
@@ -196,91 +234,99 @@
+(if_statement consequence: (statement_block ["{" "}"] @punctuation.bracket.conditional))
+(if_statement alternative: (else_clause (statement_block ["{" "}"] @punctuation.bracket.conditional)))
+
+(switch_statement body: (switch_body ["{" "}"] @punctuation.bracket.conditional))
+(switch_case ":" @punctuation.delimiter.conditional)
+(switch_default ":" @punctuation.delimiter.conditional)
+
+(for_statement ["(" ")"] @punctuation.bracket.repeat)
+(for_statement body: (statement_block ["{" "}"] @punctuation.bracket.repeat))
+(for_in_statement ["(" ")"] @punctuation.bracket.repeat)
+(for_in_statement body: (statement_block ["{" "}"] @punctuation.bracket.repeat))
+(while_statement condition: (parenthesized_expression ["(" ")"] @punctuation.bracket.repeat))
+(while_statement body: (statement_block ["{" "}"] @punctuation.bracket.repeat))
+(do_statement body: (statement_block ["{" "}"] @punctuation.bracket.repeat))
+(do_statement condition: (parenthesized_expression ["(" ")"] @punctuation.bracket.repeat))
+
+(with_statement body: (statement_block ["{" "}"] @punctuation.bracket.with))
+
+(try_statement body: (statement_block ["{" "}"] @punctuation.bracket.exception))
+(catch_clause ["(" ")"] @punctuation.bracket.exception)
+(catch_clause body: (statement_block ["{" "}"] @punctuation.bracket.exception))
+(finally_clause body: (statement_block ["{" "}"] @punctuation.bracket.exception))
+
+(class_declaration body: (class_body ["{" "}"] @punctuation.bracket.class))
+
+(function_declaration parameters: (formal_parameters ["(" ")"] @punctuation.bracket.function))
+(function_declaration parameters: (formal_parameters "," @punctuation.delimiter.function))
+(function_declaration body: (statement_block ["{" "}"] @punctuation.bracket.function))
+(generator_function_declaration parameters: (formal_parameters ["(" ")"] @punctuation.bracket.function))
+(generator_function_declaration parameters: (formal_parameters "," @punctuation.delimiter.function))
+(generator_function_declaration body: (statement_block ["{" "}"] @punctuation.bracket.function))
+(function parameters: (formal_parameters ["(" ")"] @punctuation.bracket.function))
+(function parameters: (formal_parameters "," @punctuation.delimiter.function))
+(function body: (statement_block ["{" "}"] @punctuation.bracket.function))
+(arrow_function parameters: (formal_parameters ["(" ")"] @punctuation.bracket.function))
+(arrow_function parameters: (formal_parameters "," @punctuation.delimiter.function))
+(arrow_function "=>" @punctuation.special.function)
+(arrow_function body: (statement_block ["{" "}"] @punctuation.bracket.function))
+(generator_function_declaration "*" @punctuation.special.function)
+(method_definition "*" @punctuation.special.function)
+(method_definition parameters: (formal_parameters ["(" ")"] @punctuation.bracket.function))
+(method_definition parameters: (formal_parameters "," @punctuation.delimiter.function))
+(method_definition body: (statement_block ["{" "}"] @punctuation.bracket.function))
```

Comment:
I personally think this section is a bit too much and seems like what you want to configure on your own. Probably better to let users add them instead, or only some of these should be added